lua/harpoon/buffer.lua

```diff
@@ -52,27 +52,25 @@ function M.setup_autocmds_and_keymaps(bufnr)
     vim.api.nvim_buf_set_option(bufnr, "buftype", "acwrite")
     vim.api.nvim_buf_set_option(bufnr, "bufhidden", "delete")
 
-    vim.api.nvim_buf_set_keymap(
-        bufnr,
-        "n",
-        "q",
-        "<Cmd>lua require('harpoon.buffer').run_toggle_command('q')<CR>",
-        { silent = true }
-    )
-    vim.api.nvim_buf_set_keymap(
-        bufnr,
-        "n",
-        "<ESC>",
-        "<Cmd>lua require('harpoon.buffer').run_toggle_command('Esc')<CR>",
-        { silent = true }
-    )
-    vim.api.nvim_buf_set_keymap(
-        bufnr,
-        "n",
-        "<CR>",
-        "<Cmd>lua require('harpoon.buffer').run_select_command()<CR>",
-        {}
-    )
+    vim.keymap.set("n", "q", function()
+        M.run_toggle_command("q")
+    end, { buffer = bufnr, silent = true })
+
+    vim.keymap.set("n", "<ESC>", function()
+        M.run_toggle_command("<ESC>")
+    end, { buffer = bufnr, silent = true })
+
+    vim.keymap.set("n", "<CR>", function()
+        M.run_select_command()
+    end, { buffer = bufnr })
+
+    vim.keymap.set("n", "<C-x>", function()
+        M.run_select_command({ split = true })
+    end, { buffer = bufnr })
+
+    vim.keymap.set("n", "<C-v>", function()
```

Comment:
so i don't want these as options within.  i think we need to figure out a way for the user to provide these through some hook.

as of now i do have a new event that i have added to listen for when the buffer is created.  I think I want to drive more and more of that for behavior, not in the code